src/services/member/schemas.ts

```diff
@@ -34,74 +34,91 @@ import { FILE_METADATA_DEFAULT_PAGE_SIZE, FILE_METADATA_MIN_PAGE } from './const
  */
 export const EMAIL_REGEX = '^[.-\\w]+@[.-\\w]+.[-\\w]{2,63}$';
 
-export default {
-  $id: 'https://graasp.org/members/',
-  definitions: {
-    member: {
-      type: 'object',
-      nullable: true,
-      properties: {
-        id: { type: 'string' },
-        name: { type: 'string' },
-        email: { type: 'string' },
-      },
-      additionalProperties: false,
-    },
+const memberSchema = Type.Object(
+  {
+    // Object definition
+    id: customType.UUID(),
+    name: Type.String({
+      format: 'username',
+      minLength: MIN_USERNAME_LENGTH,
+      maxLength: MAX_USERNAME_LENGTH,
+    }),
```

Comment:
Should this be a custom type or a re-usable schema or even just a variable that is copy-pasted in the openapi, but can be changed everywhere in the code at the same time.

Reply:
It will be transformed to customType.Username() in the nexts PR's !